### PR TITLE
Correctly merge default compiler options

### PR DESCRIPTION
**Pull request by BendingBender**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -91,6 +91,35 @@ When you have spread your tests over multiple files, you can store all those fil
 
 Now you can put all your test files in the `my-test-dir` directory.
 
+### Custom TypeScript config
+
+By default, `tsd-check` applies the following cofiguration:
```
Comment (SamVerschueren) [inline readme.md]: `cofiguration` -> `configuration`

File: readme.md [hunk 2]
```diff
@@ -91,6 +91,35 @@ When you have spread your tests over multiple files, you can store all those fil
...
+	"strict": true,
+	"jsx": "React",
+	"target": "ES2017",
+	// following options are set and are not overridable:
+	"moduleResolution": "Node",
+	"skipLibCheck": true
+}
+```
+
+If you wish override these options, you have the possibility to provide a custom TypeScript config to `tsd-check` by specifying it in `package.json`.
```
Comment (SamVerschueren) [inline readme.md]: > If you wish **to** override

File: source/test/test.ts [hunk 3]
```diff
@@ -58,20 +58,27 @@ test('fail if `typings` property is used instead of `types`', async t => {
...
+			message,
+			`Argument of type 'number | null' is not assignable to parameter of type 'number'.
   Type \'null\' is not assignable to type 'number'.`
+		);
+		t.is(severity, 'error');
+		t.is(line, 4);
+		t.is(column, 19);
+	}
+
+	assert(
```
Comment (SamVerschueren) [inline source/test/test.ts]: Remove the `assert` function, I'd rather be explicit per test case as currently it's just by accident that the line and column of best test fixtures are the same.

Just split both the `strict-null-checks` and `strict-null-checks-as-default-config-value` in separate tests.

File: source/test/fixtures/failure-strict-null-checks-as-default-config-value/package.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (SamVerschueren) [inline source/test/fixtures/failure-strict-null-checks-as-default-config-value/package.json]: I think we should move this to a `override-config` fixture or something so it's clear that the this fixture does that.

File: readme.md (hunk 2 above)
Comment (SamVerschueren) [reply readme.md]: This doesn't seem fixed?

Comment (BendingBender) [reply readme.md]: Sorry, forgot to push changes.

**Comment by SamVerschueren** [comment]
Woot woot 🍻

**Pull request merged**